CMakeLists.txt

```diff
@@ -3,9 +3,52 @@
 # See LICENSE.txt for the full license.
 
 cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.16.15.tar.gz"
+    SHA1 "6974c2150fc0d3b09de3ad1efcbf15d360647ffa"
+)
+
 project(dlib VERSION 19.0)
 
-add_library(dlib INTERFACE)
+if(DLIB_HEADER_ONLY)
+  add_library(dlib INTERFACE)
+else()
+  
+  #duplicating dlib's default options config
```

Comment:
I was trying to minimize the changes to dlib's CMakeLists, but you are correct this is not the way to do it so I took it out and made changes directly to dlib's CMakeLists file.